src/agent.py

```diff
@@ -80,21 +80,29 @@ def _on_deprecated_agent_relation_joined(self, event: ops.RelationJoinedEvent) -
             event.defer()
             return
 
+        enable_websocket = bool(self.state.remoting_config.enable_websocket)
         self.charm.unit.status = ops.MaintenanceStatus("Adding agent node.")
         try:
             jenkins.add_agent_node(
                 agent_meta=agent_meta,
                 container=container,
                 # mypy doesn't understand that host can no longer be None.
                 host=host,
+                enable_websocket=enable_websocket,
             )
             secret = jenkins.get_node_secret(container=container, node_name=agent_meta.name)
         except jenkins.JenkinsError as exc:
             self.charm.unit.status = ops.BlockedStatus(f"Jenkins API exception. {exc=!r}")
             return
 
+        configured_remoting_external_url = self.state.remoting_config.external_url
+        jenkins_url = (
+            f"http://{host}:{jenkins.WEB_PORT}"
```

Comment:
A less verbose alternative (not sure if better):

`configured_remoting_external_url || f"http://{host}:{jenkins.WEB_PORT}"`